### PR TITLE
Reduce dependency on the `signal` property of `SignalType`

### PR DESCRIPTION
**Pull request by dapenggao**
In many places `signal` can actually be replaced by `self`, which can give rise to looser coupling. 
Other changes are taken from #2386.
All tests are passed.
PS: The indentation changes are actually replacing spaces with tabs.

### CODE REVIEW

File: ReactiveCocoa/Swift/ObjectiveCBridging.swift [hunk 1]
```diff
@@ -119,7 +119,7 @@ public func toRACSignal<T: AnyObject, E: NSError>(producer: SignalProducer<T?, E
 				subscriber.sendError(error)
 			case .Completed:
 				subscriber.sendCompleted()
-			default:
+			case .Interrupted:
```
Comment (NachoSoto) [inline ReactiveCocoa/Swift/ObjectiveCBridging.swift]: :thumbsup:

**Comment by NachoSoto** [comment]
Looks good!

Could you separate the indentation change though?

File: ReactiveCocoa/Swift/Signal.swift [hunk 2]
```diff
@@ -311,11 +311,11 @@ extension SignalType where Value: SignalProducerType, Error == Value.Error {
 	/// emitted from `signal` complete.
 	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
 	private func concat() -> Signal<Value.Value, Error> {
-		return Signal<Value.Value, Error> { [signal = self.signal] observer in
+		return Signal<Value.Value, Error> { observer in
```
Comment (iv-mexx) [inline ReactiveCocoa/Swift/Signal.swift]: I'm not sure about this - doesn't this now capture `self` with a strong reference (thus leading to a potential strong reference cycle), whereas it previously had captured only `signal` and not `self`?

Comment (andersio) [inline ReactiveCocoa/Swift/Signal.swift]: Both captures the same reference. `Signal`'s implementation of `SignalType.signal` just returns a reference of the instance itself.

Comment (dapenggao) [inline ReactiveCocoa/Swift/Signal.swift]: @iv-mexx `self` is the same object as `self.signal` at runtime. So capturing either `self.signal` or `self` will have the same effect.

File: ReactiveCocoa/Swift/ObjectiveCBridging.swift [hunk 3]
```diff
@@ -167,7 +167,7 @@ public func toRACSignal<Value: AnyObject, Error: NSError>(signal: Signal<Value?,
                 subscriber.sendError(error)
             case .Completed:
                 subscriber.sendCompleted()
-            default:
+			case .Interrupted:
```
Comment (dapenggao) [inline ReactiveCocoa/Swift/ObjectiveCBridging.swift]: This line displays perfectly well on Xcode, I don't know why it got shifted on GitHub...

File: ReactiveCocoa/Swift/Signal.swift (hunk 2 above)
Comment (iv-mexx) [inline ReactiveCocoa/Swift/Signal.swift]: Ok, thanks for the clarification :+1:

File: ReactiveCocoa/Swift/ObjectiveCBridging.swift (hunk 3 above)
Comment (iv-mexx) [inline ReactiveCocoa/Swift/ObjectiveCBridging.swift]: Tabs vs. Spaces.
The old line was indented wit a total of 12 spaces, the new line is indented with 3 tabs.

**Comment by dapenggao** [comment]
`observeWithStates` is changed to a private method for better consistency with `observe`.

File: ReactiveCocoa/Swift/ObjectiveCBridging.swift (hunk 3 above)
Comment (NachoSoto) [inline ReactiveCocoa/Swift/ObjectiveCBridging.swift]: In any case, it's always a good idea to not mix indentation changes with other functional changes to make reviewing code easier. I know about the `w=0` flag, but still, it's a good idea to keep the git history clean :)

**Comment by NachoSoto** [comment]
In any case, it's always a good idea to not mix indentation changes with other functional changes to make reviewing code easier. I know about the w=0 flag, but still, it's a good idea to keep the git history clean :)

In particular, it's going to be really hard to look back at the history and figure out what changed in here:
screen shot 2015-10-01 at 12 09 11

Could you amend the commits to undo the indentation changes?

**Comment by dapenggao** [comment]
@NachoSoto So in that particular case, the entire function is shifted because it became nested in an extension.

**Comment by NachoSoto** [comment]
> But in the case above, the entire function is shifted because it became nested in an extension.

Oh my bad, makes sense :+1:

**Pull request merged**